### PR TITLE
Update Okta Org Cleaner

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
## Description
Updated Okta Org cleaner with new things to clean up.

## Category

- [ ] Bugfix
- [x] Enhancement
- [ ] New Feature
- [ ] Configuration Change
- [ ] Versioning Change
- [ ] Unit Test(s)
- [ ] Documentation

### CODE REVIEW

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy [hunk 1]
```diff
@@ -60,27 +63,75 @@ class OktaOrgCleaner {
...
+
+        log.info("Deleting IdentityProviders:")
+        client.listIdentityProviders().stream()
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting SmsTemplates:")
+        client.listSmsTemplates().stream()
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:134]: Great addition, but you probably want to filter each of these streams.
For example, this logic _might_ delete the `default` authorization server

Comment (arvindkrishnakumar-okta) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:134]: I checked this and it does not delete the default Authorization server.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy [hunk 2]
```diff
@@ -60,27 +63,77 @@ class OktaOrgCleaner {
...
+        log.info("Deleting IdentityProviders:")
+        client.listIdentityProviders().stream()
+            .filter()
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting SmsTemplates:")
+        client.listSmsTemplates().stream()
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:136]: The filtering should be done similar to the above code blocks were it delete the test created objects only.
The assumption here is that, anything created by a human would NOT have the format of `policy\\+${uuidRegex}.*` and anything created by a test would.  There are many things you could have created in your org and you may not want to delete them all.

Also, ideally, the test SHOULD clean up everything it creates. But even in a base case scenario this util class is useful for:
* Fixing tests that are not doing proper cleanup
* Cleaning up after tests under development
* Any random test failure that could have left resources behind.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy [hunk 3]
```diff
@@ -60,27 +63,81 @@ class OktaOrgCleaner {
...
         client.listPolicies(PolicyType.OKTA_SIGN_ON.toString()).stream()
-            .filter { it.getName().matches("policy\\+${uuidRegex}.*")}
+            .filter { it.getName().matches("policy\\+${uuidRegex}.*") }
+            .forEach {
+                it.delete()
+            }
+
+        log.info("Deleting LinkedObjectDefinitions:")
+        client.listLinkedObjectDefinitions().stream()
+            .filter { it.getName().startsWith("manager") }
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:93]: same problem as before, if the test uses a pattern that could be a _real_ usage, the test should be updated (for example, if you need a _short_ name (for cases where the field is not long enough for a UUID, you could use `SDK-IT-*` or something, just update the IT in this PR

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy [hunk 4]
```diff
@@ -60,27 +63,81 @@ class OktaOrgCleaner {
...
+        client.listEventHooks().stream()
+            .filter { it.getName().matches(".*-${uuidRegex}.*") }
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting UserTypes:")
+        client.listUserTypes().stream()
+            .filter { !it.getDefault() && it.getName().startsWith("java_sdk_user_type_") }
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:117]: +1 this is a good example of my comment on the "manager" line above

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy [hunk 5]
```diff
@@ -60,27 +63,81 @@ class OktaOrgCleaner {
...
+            .forEach {
+                it.delete()
+            }
+
+        log.info("Deleting AuthorizationServers:")
+        client.listAuthorizationServers().stream()
+            .filter { !it.getName().equals("default") && it.getName().startsWith("java-sdk-authorization-server-") }
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:124]: nit: the previous example uses snake case, this one uses kabob case

My personal pref is to add the term `it` or `tests` in there, but this is only a suggestion

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy [hunk 6]
```diff
@@ -60,27 +63,81 @@ class OktaOrgCleaner {
...
+            .forEach {
+                it.deactivate()
+                it.delete()
+            }
+
+        log.info("Deleting IdentityProviders:")
+        client.listIdentityProviders().stream()
+            .filter { it.getName().startsWith("Mock") }
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:132]: see the above comment on naming.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy (hunk 3 above)
Comment (arvindkrishnakumar-okta) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy:93]: sure, will update the PR once I refactor all the test object names.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/LinkedObjectsIT.groovy [hunk 7]
```diff
@@ -35,8 +35,8 @@ class LinkedObjectsIT extends ITSupport {
 
     @Test
     void addLinkedObjectDefinitionTest() {
-        String primaryName = "manager" + RandomStringUtils.randomAlphanumeric(25)
-        String associatedName = "subordinate" + RandomStringUtils.randomAlphanumeric(25)
+        String primaryName = "java_sdk_it_" + RandomStringUtils.randomAlphanumeric(25)
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/LinkedObjectsIT.groovy:38]: `LinkedObject` creation does not support `-` in name.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/UserTypesIT.groovy [hunk 8]
```diff
@@ -34,7 +34,7 @@ class UserTypesIT extends ITSupport {
 
     @Test
     void createUserTypeTest() {
-        String name = "java_sdk_user_type_" + RandomStringUtils.randomAlphanumeric(15)
+        String name = "java_sdk_it_" + RandomStringUtils.randomAlphanumeric(15)
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/UserTypesIT.groovy:37]: `UserType` creation does not support `-` in name. Therefore, resorted to snake case here.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy (hunk 5 above)
Comment (arvindkrishnakumar-okta) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy]: addressed this.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy (hunk 6 above)
Comment (arvindkrishnakumar-okta) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/util/OktaOrgCleaner.groovy]: sure, addressed.

File: examples/quickstart/src/main/java/quickstart/Quickstart.java [hunk 9]
```diff
@@ -105,6 +111,20 @@ public static void main(String[] args) {
...
             throw e;
         }
+        finally {
+            // cleanup
+
+            // deactivate, delete user
+            if (user != null && user.getStatus() != UserStatus.DEPROVISIONED) {
+                user.deactivate();
+            }
+            user.delete();
```
Comment (bdemers) [inline examples/quickstart/src/main/java/quickstart/Quickstart.java:121]: If user is null here this delete will fail

Comment (arvindkrishnakumar-okta) [reply examples/quickstart/src/main/java/quickstart/Quickstart.java:121]: addressed this.

**Pull request merged**